proposals/20230323-end-to-end-testing.md

```diff
@@ -0,0 +1,372 @@
+test for the Manifest repo, this doc proposes a design to set up the infrastructure needed to run the necessary tests.
+
+References
+
+- [Optional Test Infra Deprecation Notice](https://github.com/kubeflow/testing/issues/993)
+- [Alternative solution to removal of test on optional-test-infra](https://github.com/kubeflow/testing/issues/1006)
+
+## Goal
+
+Enable the e2e testing for the Manifest repo and leverage it to shorten the manifest testing phase of the Kubeflow
+release cycle and to increase quality of the Kubeflow release by ensuring Kubeflow components and dependencies work
+correctly together.
+
+## Proposal
+
+After some initial conversations, it has been agreed to create integration tests based on GitHub Actions, which will
+spawn an EC2 instance with enough resources to deploy the complete Kubeflow solution and run some end-to-end testing.
+
+### Alternative solutions considered
+
+#### Prow
+
+While there are some existing tests with Prow, those tests were discarded due to them not having been updated in 2 years
+and there being a high amount of complexity in these tests. After some investigation, the Manifests Working Group
+decided that it would be more work adapting those tests to the current state of manifests than starting from scratch
+with lower complexity.
+
+#### Self-hosted runners
+
+Self-hosted runners are not recommended with public repositories due to security concerns with how it behaves on a pull
+request made by a forked repository.
+
+## Implementation
+
+Below lists steps the GitHub actions will perform to complete end-to-end testing
+
+- [Create an EC2 instance](#create-an-ec2-instance)
+- [Install a Kubernetes on the instance](#install-a-kubernetes-on-the-instance)
+- [Deploy Kubeflow](#deploy-kubeflow)
+- [Run tests](#run-tests)
+- [Log and report errors](#log-and-report-errors)
+- [Clean up](#clean-up)
+
+### Create an EC2 instance
+
+Access the AWS credentials (stored as GH Secrets) and create an EC2 instance
+
+Using [juju](https://juju.is/) as an orchestration, configure AWS credentials and deploy an EC2 instance with the
+following configurations
+
+- Image: Ubuntu Server (latest)
+- Type: t3a.xlarge
+- Root disk: 80G
+- Region: us-east-1 (default)
+
+#### Why juju?
+
+Juju allows easy configuration to various cloud providers. In the future, if there comes a reason to shift to another
+infrastructure provider, it would allow us to pivot quickly.
+
+While juju provides more capability, the proposal is to use the tool as config management and a medium to deploy and
+connect with EC2 instances.
+
+**Note**: Using GitHub Secrets to store AWS credentials will not allow any forked repositories to access the secrets.
+
+### Install a Kubernetes on the Instance
+
+Install Kubernetes on the EC2 instance where Kubeflow will be deployed and tested
+
+To install Kubernetes, we explored two options and propose to use **KinD**
```

Comment:
Following my rationale abovee, let's only keep this section for how to install KinD and put MicroK8s in the Alternatives Considered section at the end

Reply:
Moved to the last section